src/Akeneo/Category/back/tests/Specification/Infrastructure/Builder/TemplateBuilderSpec.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Specification\Akeneo\Category\Infrastructure\Builder;
+
+use Akeneo\Category\Domain\Model\Category;
+use Akeneo\Category\Domain\Query\GetCategoryInterface;
+use Akeneo\Category\Domain\ValueObject\CategoryId;
+use Akeneo\Category\Domain\ValueObject\Code;
+use Akeneo\Category\Domain\ValueObject\LabelCollection;
+use Akeneo\Category\Infrastructure\Builder\TemplateBuilder;
+use PhpSpec\ObjectBehavior;
+
+/**
+ * @copyright 2022 Akeneo SAS (https://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class TemplateBuilderSpec extends ObjectBehavior
+{
+    public function let(GetCategoryInterface $getCategory): void
+    {
+        $this->beConstructedWith($getCategory);
+    }
+
+    function it_is_initializable(): void
+    {
+        $this->shouldHaveType(TemplateBuilder::class);
+    }
+
+    public function it_generate_a_template_from_a_category_code(
+        Code $searchingCategoryCode,
+        GetCategoryInterface $getCategory,
+        Category $categoryTree,
+        Code $categoryTreeCode,
+        CategoryId $categoryId
+    )
+    {
+        $searchingCategoryCode->__toString()->willReturn('category_code');
```

Comment:
In this case, it could be simpler to use directly the instance of the Code values object instead of using mocks. Moreover, all validation or business rules will be available

```suggestion
        const $searchingCategoryCode = new Code('category_code');
```

Reply:
It simplified a lot the tests, thx